### PR TITLE
ENH: Period comparisons with mismatched freq use py3 behavior

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/arithmetic/test_period.py [hunk 1]
```diff
@@ -354,12 +354,13 @@ def test_pi_cmp_nat_mismatched_freq_raises(self, freq):
         idx1 = PeriodIndex(["2011-01", "2011-02", "NaT", "2011-05"], freq=freq)
 
         diff = PeriodIndex(["2011-02", "2011-01", "2011-04", "NaT"], freq="4M")
-        msg = "Input has different freq=4M from Period(Array|Index)"
-        with pytest.raises(IncompatibleFrequency, match=msg):
+        msg = rf"Invalid comparison between dtype=period\[{freq}\] and PeriodArray"
```
Comment (jreback) [inline pandas/tests/arithmetic/test_period.py:357]: prob should either rename the tests slightly or separate out the == case (and test != as well)

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -1541,6 +1541,10 @@ cdef class _Period(PeriodMixin):
     def __richcmp__(self, other, op):
         if is_period_object(other):
             if other.freq != self.freq:
+                if op == Py_EQ:
+                    return False
+                if op == Py_NE:
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx:1546]: could be elif

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. minor comments. ping on green.

File: pandas/_libs/tslibs/period.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/period.pyx:1546]: sure

File: pandas/tests/arithmetic/test_period.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_period.py:357]: will rename for now.  i think we may be able to re-use a helper from the dt64 tests, will take a look before long

**Review by jreback — APPROVED** [review]

**Pull request merged**